src/Middleware/WebSockets/src/HandshakeHelpers.cs

```diff
@@ -72,5 +75,89 @@ public static string CreateResponseKey(string requestKey)
+                        options.ClientContextTakeover = false;
+                        builder.Append("; ").Append(WebSocketDeflateConstants.ClientNoContextTakeover);
+                    }
+                    else if (value.SequenceEqual(WebSocketDeflateConstants.ServerNoContextTakeover))
+                    {
+                        options.ServerContextTakeover = false;
+                        builder.Append("; ").Append(WebSocketDeflateConstants.ServerNoContextTakeover);
+                    }
+                    else if (value.StartsWith(WebSocketDeflateConstants.ClientMaxWindowBits))
+                    {
+                        var clientMaxWindowBits = ParseWindowBits(value, WebSocketDeflateConstants.ClientMaxWindowBits);
+                        if (clientMaxWindowBits > options.ClientMaxWindowBits)
+                        {
+                            return false;
+                        }
+                        // if client didn't send a value for ClientMaxWindowBits use the value the server set
+                        options.ClientMaxWindowBits = clientMaxWindowBits ?? options.ClientMaxWindowBits;
+                        builder.Append("; ").Append(WebSocketDeflateConstants.ClientMaxWindowBits).Append('=')
+                            .Append(options.ClientMaxWindowBits.ToString(CultureInfo.InvariantCulture));
+                    }
+                    else if (value.StartsWith(WebSocketDeflateConstants.ServerMaxWindowBits))
+                    {
+                        var serverMaxWindowBits = ParseWindowBits(value, WebSocketDeflateConstants.ServerMaxWindowBits);
+                        if (serverMaxWindowBits > options.ServerMaxWindowBits)
+                        {
+                            return false;
+                        }
+                        // if client didn't send a value for ServerMaxWindowBits use the value the server set
+                        options.ServerMaxWindowBits = serverMaxWindowBits ?? options.ServerMaxWindowBits;
+
+                        builder.Append("; ")
+                            .Append(WebSocketDeflateConstants.ServerMaxWindowBits).Append('=')
+                            .Append(options.ServerMaxWindowBits.ToString(CultureInfo.InvariantCulture));
+                    }
+
+                    static int? ParseWindowBits(ReadOnlySpan<char> value, string propertyName)
+                    {
+                        var startIndex = value.IndexOf('=');
+
+                        // parameters can be sent without a value by the client, we'll use the values set by the app developer or the default of 15
+                        if (startIndex < 0)
+                        {
+                            return null;
+                        }
+
+                        if (!int.TryParse(value[(startIndex + 1)..], NumberStyles.Integer, CultureInfo.InvariantCulture, out int windowBits) ||
+                            windowBits < 9 ||
```

Comment:
The spec says 8 is the lower bound. There was an issue with the implementation not supporting that, right? Even so, it shouldn't generate an exception here, it should be politely ignored (ParseDeflateOptions returns false). Include a comment that 8 is a valid value but unsupported.